src/main/java/de/tum/in/www1/artemis/web/rest/ModelingAssessmentResource.java

```diff
@@ -85,23 +85,21 @@ public ModelingAssessmentResource(AuthorizationCheckService authCheckService, Us
      * PUT modeling-submissions/:submissionId/assessment : save manual modeling assessment. See {@link AssessmentResource#saveAssessment}.
      *
      * @param submissionId id of the submission
+     * @param resultId id of the result
      * @param feedbacks list of feedbacks
      * @param submit if true the assessment is submitted, else only saved
      * @return result after saving/submitting modeling assessment
      */
     @ResponseStatus(HttpStatus.OK)
     @ApiResponses({ @ApiResponse(code = 200, message = PUT_SUBMIT_ASSESSMENT_200_REASON, response = Result.class),
             @ApiResponse(code = 403, message = ErrorConstants.REQ_403_REASON), @ApiResponse(code = 404, message = ErrorConstants.REQ_404_REASON) })
-    @PutMapping("/modeling-submissions/{submissionId}/assessment")
+    @PutMapping("/modeling-submissions/{submissionId}/result/{resultId}/assessment")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<Result> saveModelingAssessment(@PathVariable long submissionId, @RequestParam(value = "submit", defaultValue = "false") boolean submit,
-            @RequestBody List<Feedback> feedbacks) {
+    public ResponseEntity<Result> saveModelingAssessment(@PathVariable long submissionId, @PathVariable long resultId,
+            @RequestParam(value = "submit", defaultValue = "false") boolean submit, @RequestBody List<Feedback> feedbacks) {
         Submission submission = submissionService.findOneWithEagerResultAndFeedback(submissionId);
         ModelingExercise exercise = (ModelingExercise) submission.getParticipation().getExercise();
 
-        // if a result exists, we want to override it, otherwise create a new one
-        var resultId = submission.getLatestResult() != null ? submission.getLatestResult().getId() : null;
-
         ResponseEntity<Result> response = super.saveAssessment(submission, submit, feedbacks, resultId);
```

Comment:
One additional question to the findings above: Doesn't this need validation for if the current user has access to the course?

Reply:
Yes!

Reply:
Where are we currently using such check? So that I can take a look into it

Reply:
Hopefully everywhere. Should be userservice.isatleast..

Reply:
The authorization checks are done inside the `saveAssessment` method of the super class AssessmentResource

Reply:
True, good catch @anditurdiu. This resolves this comment. No need for further checks @nicolasruscher.

Reply:
Yes, what @anditurdiu said, it can be found within the AssessmentResource within: `saveAssessment`